crates/sui-framework/packages/bridge/sources/committee.move

```diff
@@ -164,9 +165,35 @@ module bridge::committee {
             registration
         };
 
+        // check uniqeuness of the bridge pubkey.
+        // `try_create_next_committee` will abort if bridge_pubkey_bytes are not unique and
+        // that will fail the end of epoch transaction (possibly "forever", well, we
+        // need to deploy proper validator changes to stop end of epoch from failing).
+        assert!(check_uniqueness_bridge_keys(self, bridge_pubkey_bytes), EDuplicatePubkey);
+
         emit(registration)
     }
 
+    // Return true if `bridge_pubkey_bytes` is unique in the committee.
+    // Dupicate keys would cause `try_create_next_committee` to fail and,
+    // in consequence, an end of epoch transaction to fail (safe mode run).
+    // This check will ensure the creation of the committee is correct.
+    fun check_uniqueness_bridge_keys(self: &BridgeCommittee, bridge_pubkey_bytes: vector<u8>): bool {
+        let mut count = self.member_registrations.size();
+        let mut found = false;
+        while (count > 0) {
+            count = count - 1;
+            let (_, registration) = self.member_registrations.get_entry_by_idx(count);
+            if (registration.bridge_pubkey_bytes == bridge_pubkey_bytes) {
+                if (found) {
+                    return false
+                };
+                found = true;
```

Comment:
nit: I'd probably pass down the sender's address too and check as the "found" thing messed with my brain (but could be me malfunctioning)

`if (registration.bridge_pubkey_bytes == bridge_pubkey_bytes && registration.sui_address != sender
 ) { 
  return false 
 }`

Reply:
not sure I follow, the found is to make sure that `bridge_pubkey_bytes` is unique in the list of proposed committee upgrade.
I may be missing something here

Reply:
it messed with my brain as well and could be me malfunctioning too.

Reply:
renamed the field and added comments that should hopefully help fix the confusion